src/database/mongo.js

```diff
@@ -111,38 +112,46 @@
 					if (err) {
 						return callback(err);
 					}
-					createSessionStore();					
+					callback();
 				});
 			} else {
 				winston.warn('You have no mongo password setup!');
-				createSessionStore();
-			}
-
-			function createSessionStore() {
-				var sessionStore;
-				if (nconf.get('redis')) {
-					sessionStore = require('connect-redis')(session);
-					var rdb = require('./redis');
-					rdb.client = rdb.connect();
-
-					module.sessionStore = new sessionStore({
-						client: rdb.client,
-						ttl: 60 * 60 * 24 * 14
-					});
-				} else if (nconf.get('mongo')) {
-					sessionStore = require('connect-mongo')(session);
-					module.sessionStore = new sessionStore({
-						db: db
-					});
-				}
 				callback();
-			}			
+			}
 		});
 	};
 
+	module.initSessionStore = function (callback) {
+		var meta = require('../meta');
+		var sessionStore;
+		if (nconf.get('redis')) {
+			sessionStore = require('connect-redis')(session);
+			var rdb = require('./redis');
+			rdb.client = rdb.connect();
+
+			var ttlDays = 1000 * 60 * 60 * 24 * (parseInt(meta.config.loginDays, 10) || 0);
+			var ttlHours = 1000 * 60 * 60 * (parseInt(meta.config.loginHours, 10) || 0);
+			var ttlMinutes = 1000 * 60 * (parseInt(meta.config.loginMinutes, 10) || 0);
+			var ttlSeconds = 1000 * (parseInt(meta.config.loginSeconds, 10) || 0);
+			var ttlTotal = (ttlDays + ttlHours + ttlMinutes + ttlSeconds) || 1209600000; // Default to 14 days
```

Comment:
I'm not convinced that separate configurations for days, hours, minutes, and seconds are required, nor are they the _best_ approach for this.

If the end result is to save a calculated `ttl`, perhaps it is easiest to leave `loginDays` as is, but if additional granularity is required, then an additional field `loginSeconds` supercedes `loginDays`.

For example, if I wanted my cookie timeout to be 10 days, 4 hours, 6 minutes, and 59 seconds ( 😧  ), this approach would work, but I think we can both agree that that isn't a common use case.

So let's try a troublesome client who wants a timeout of "two-thirds of a week". We could enter it as:

* `ttl` = (4 days, 14 hours, 52 minutes, and 48 seconds) * 1000, or
* `ttl` = 399168 seconds * 1000

Probably easier to just do the latter, no? 😄